emsdk.py

```diff
@@ -1440,6 +1440,54 @@ def emscripten_npm_install(tool, directory):
     errlog('Error running %s:\n%s' % (e.cmd, e.output))
     return False
 
+  # Manually install the appropriate native Closure Compiler package
+  # This is currently needed because npm ci will install the packages
+  # for Closure for all platforms, adding 180MB to the download size
+  # There are two problems here:
+  #   1. npm ci does not consider the platform of optional dependencies
+  #      https://github.com/npm/cli/issues/558
+  #   2. A bug with the native compiler has bloated the packages from
+  #      30MB to almost 300MB
+  #      https://github.com/google/closure-compiler-npm/issues/186
+  # If either of these bugs are fixed then we can remove this exception
+  # See also https://github.com/google/closure-compiler/issues/3925
+  closure_compiler_native = ''
+  if LINUX and ARCH in ('x86', 'x86_64'):
+    closure_compiler_native = 'google-closure-compiler-linux'
+  if MACOS and ARCH in ('x86', 'x86_64'):
+    closure_compiler_native = 'google-closure-compiler-osx'
+  if WINDOWS and ARCH == 'x86_64':
+    closure_compiler_native = 'google-closure-compiler-windows'
+
+  if closure_compiler_native:
+    closure_compiler_native += '@20220104.0.0'
```

Comment:
Won't this need updating every time we do an npm update on the emscripten-side?   (iirc we have updated several times over the years so any one value here might be work).

Reply:
Yes it does, and that is quite unfortunate. I wrote a little bit about that earlier in https://github.com/emscripten-core/emsdk/issues/802#issuecomment-829095886 . I don't see another way but to keep these in sync.

Reply:
Can we maybe do something like this:

```
closure_version = json.load('package.json')['dependencies']['google-closure-compiler']
```